runtime/vm/classsupport.c

```diff
@@ -226,6 +226,20 @@ internalCreateArrayClass(J9VMThread* vmThread, J9ROMArrayClass* romClass, J9Clas
 		LOAD_LOCATION_UNKNOWN,
 		NULL,
 		NULL);
+
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+	/* When creating an array of valuetype elements, the array elements are initialized to the defaultValue of the
+	 * element type. As a result the element type must be fully initialized (if its a valuetype) before creating an
+	 * instance of the array.
+	 */
+	if ((NULL != result) && J9_IS_J9CLASS_VALUETYPE(elementClass)) {
+		UDATA initStatus = elementClass->initializeStatus;
+		if ((J9ClassInitSucceeded != initStatus) && ((UDATA)vmThread != initStatus)) {
+			initializeClass(vmThread, elementClass);
```

Comment:
If this fails, you should probably return NULL (presumably the caller already has to handle any pending exceptions). Check pending exception here and set result to NULL (and re-NULL the arrayClass field). I'm not 100% sure, but I suspect we should only be attempting